include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -102,19 +104,47 @@ class Cube : public std::enable_shared_from_this<Cube> {
 
     /// Is the current cube root.
     [[nodiscard]] bool is_root() const noexcept;
+    /// Is the current cube a leaf.
+    [[nodiscard]] bool is_leaf() const noexcept;
     /// At which child level this cube is.
     /// root cube = 0
     [[nodiscard]] std::size_t grid_level() const noexcept;
     /// Count the number of Type::SOLID and Type::NORMAL cubes.
     [[nodiscard]] std::size_t count_geometry_cubes() const noexcept;
 
+    [[nodiscard]] glm::vec3 center() const noexcept {
+        return m_position + 0.5f * m_size;
+    }
+
+    [[nodiscard]] const glm::vec3 &position() const noexcept {
```

Comment:
```suggestion
    [[nodiscard]] glm::vec3 position() const noexcept {
```
We return by value for other getters too.

Reply:
Right. Done.

Reply:
Huh?
> We return by value for other getters too.

Where?

Reply:
Right. Maybe in cube.hpp we should turn all get methods of composed types into const reference get methods?

Reply:
On the other side why const reference? I mean the cube can be deleted in the meantime, if used later.

Reply:
Hmm good point.

Reply:
There's no need to copy the `glm::vec3`, so we should return a const reference since the compiler can easily remove the reference if it wants to, but can't re-add it. We should only return small primitives like `float` by value.

Reply:
And what about thread safety?

Reply:
For a `glm::vec3`?

Reply:
If you have a reference to the cube to access the position in the first place, then the returned `glm::vec3` will always be valid I think.

Reply:
I will keep the commit which changes it to a const reference in now.